Source/1.3/Windows/SettlementPlacementWindow.cs

```diff
@@ -0,0 +1,247 @@
+                }
+
+                rectFull = rectFull.MoveRect(new Vector2(0f, 29f));
+                count++;
+            }
+
+            Widgets.EndScrollView();
+        }
+
+        private void DrawTop()
+        {
+            Text.Font = GameFont.Medium;
+            Widgets.Label(rectTop, "Empire_SPW_Title".Translate());
+            Text.Font = GameFont.Small;
+
+            Widgets.DrawLineHorizontal(rectTop.x, rectTop.yMax, rectTop.width);
+        }
+
+        private void DrawBottom()
+        {
+            bool canPlace = CanPlaceHere(out List<string> reasons, playerEmpire);
+
+            Widgets.DrawLineHorizontal(rectBot.x, rectBot.y, rectBot.width);
+            TooltipHandler.TipRegion(rectButtonApply, reasons.Join(newString => newString, "\n"));
+
+            if (!canPlace) GUI.color = new Color(1f, 0.4f, 0.4f);
+            rectButtonApply.DrawButtonText("Empire_SPW_Apply".Translate(), ApplyAction);
+            GUI.color = Color.white;
+        }
+
+        private void GetTileData()
+        {
+            int newTile = Find.WorldSelector.selectedTile;
+            if (newTile == -1 || newTile == selectedWorldTile) return;
+            selectedWorldTile = newTile;
+
+            foreach (ResourceDef def in DefDatabase<ResourceDef>.AllDefsListForReading)
+            {
+                tileMods.SetOrAdd(def, def.GetTileModifier(Find.WorldGrid.tiles[selectedWorldTile]));
+            }
+        }
+
+        public bool CanPlaceHere(out List<string> reasons, Empire playerEmpire)
+        {
+            bool flag = true;
+            reasons = new List<string>();
+            List<Tile> tiles = Find.WorldGrid.tiles;
+
+            if (selectedWorldTile > tiles.Count || selectedWorldTile == -1)
+            {
+                reasons.Add("Empire_SPW_TileOutOfRange".Translate());
+                return false; //Not just change the flag here because the next line would error
+            }
+
+            Tile tile = tiles[selectedWorldTile];
+
+            Territory territory = TerritoryManager.GetTerritoryManager.GetTerritory(Faction.OfPlayer);
+            if (tile.WaterCovered)
+            {
+                reasons.Add("Empire_SPW_Water".Translate());
+                flag = false;
+            }
+
+            if (!playerEmpire.StorageTracker.CanRemoveThingsFromStorage(Empire.SettlementCost))
+            {
+                reasons.Add("Empire_SPW_NoResources".Translate());
+                flag = false;
+            }
+
+            if (!territory.HasTile(selectedWorldTile))
+            {
+                reasons.Add("Empire_SPW_TileNotInTerritory".Translate());
+                flag = false;
+            }
+
+            return flag;
+        }
+
+        public void ApplyAction()
+        {
+            if (!CanPlaceHere(out List<string> reasons, playerEmpire))
+            {
+                Messages.Message(new Message(string.Join("\n", reasons), MessageTypeDefOf.RejectInput));
```

Comment:
Multiline Messages look weird and overlap on the screen.
Consider replacing with a popup window, maybe?

Reply:
I forgor

Reply:
[17d3c62](https://github.com/Danimineiro/Empire-Rewritten/pull/30/commits/17d3c6267516fc359cf929b7db61858461b127a5) fixes